### PR TITLE
Rename: service => network & entity => instance

### LINKED ISSUE #30: Change: Document/Entity -> Instance
This change is needed to match with go implementation. It shouldn't affect implementation details or anything beyond aesthetics, but will improve the developer experience to have a unified naming convention. After this change, the naming should be updated in the white paper.

### LINKED ISSUE #35: Rename Service to Network/Net
To match Go versions and naming conventions. Note that with this change, the white paper should also be updated.

### PR DESCRIPTION
**Pull request by carsonfarmer**
Signed-off-by: Carson Farmer <carson.farmer@gmail.com>

## Description

Rename Service to Network, and Entity to Instance throughout. Note that we're leaving Document 'as is' because it is simply a wrapper around a Collection/Instance.

Fixes #35 
Fixes #30 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

All existing tests pass, and updates were made to reflect the name changes.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~I have made corresponding changes to the documentation~
- [x] My changes generate no new warnings
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/database/src/collection.ts [hunk 1]
```diff
@@ -263,24 +261,24 @@ export class Collection<T extends Entity = any> extends ReadonlyCollection<T> {
   /**
    * Insert (multiple) new entities.
```
Comment (andrewxhill) [inline packages/database/src/collection.ts]: minor

File: packages/network/src/network.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ export class Service implements Interface {
    * @param id The Thread id.
```
Comment (andrewxhill) [inline packages/network/src/network.ts]: ID. Wouldn't have wasted the time of a comment, but it got me wondering if we should use ThreadID as convention when describing this particular ID or if it is the "ID of the Thread". Dunno...

Comment (andrewxhill) [reply packages/network/src/network.ts]: Nah, it's the ID of the Thread. Just thinking of it in the context of instance ID etc.

Comment (carsonfarmer) [reply packages/network/src/network.ts]: Well, this is indeed a good question IMO. There are a lot of IDs, and the _interface_ is called ThreadID, and also ThreadKey, and maybe even ThreadAuth soon. So maybe good idea to use those names?

Comment (carsonfarmer) [reply packages/network/src/network.ts]: I'll leave that decision up to a future PR.

File: packages/database/src/collection.ts (hunk 1 above)
Comment (carsonfarmer) [reply packages/database/src/collection.ts]:
```suggestion
   * Insert (multiple) new instances.
```

**Pull request merged**